polaris/polaris/deposit/views.py

```diff
@@ -141,50 +139,69 @@ def confirm_transaction(account: str, request: Request) -> Response:
 @renderer_classes([TemplateHTMLRenderer])
 def interactive_deposit(request: Request) -> Response:
     """
-    `GET /transactions/deposit/webapp` opens a form used to input information
-    about the deposit. This creates a corresponding transaction in our
-    database, pending processing by the external agent.
     """
     # Validate query parameters: account, asset_code, transaction_id.
     account = request.GET.get("account")
     asset_code = request.GET.get("asset_code")
     asset = Asset.objects.filter(code=asset_code).first()
     transaction_id = request.GET.get("transaction_id")
     if not account:
-        return render_error_response("no 'account' provided", content_type="text/html")
+        err_msg = "no 'account' provided"
+        return render_error_response(err_msg, content_type="text/html")
     elif not (asset_code and asset):
-        return render_error_response("invalid 'asset_code'", content_type="text/html")
+        err_msg = "invalid 'asset_code'"
+        return render_error_response(err_msg, content_type="text/html")
     elif not transaction_id:
-        return render_error_response("no 'transaction_id' provided", content_type="text/html")
+        err_msg = "no 'transaction_id' provided"
+        return render_error_response(err_msg, content_type="text/html")
 
+    # Ensure the transaction exists
     try:
         transaction = Transaction.objects.get(
-            id=transaction_id, asset=asset
+            id=transaction_id,
+            asset=asset
         )
-    except Transaction.DoesNotExist:
+    except Transaction.objects.DoesNotExist:
         return render_error_response(
             "Transaction with ID and asset_code not found",
             content_type="text/html",
             status_code=status.HTTP_404_NOT_FOUND
         )
 
     if request.method == "GET":
-        form = DepositForm()
+        form = rdi.form_for_transaction(transaction)()
         return Response({"form": form}, template_name="deposit/form.html")
 
-    form = DepositForm(request.POST)
-    form.asset = asset
-    # If the form is valid, we create a transaction pending external action
-    # and render the success page.
+    # request.method == "POST"
+    form = rdi.form_for_transaction(transaction)(request.POST)
+    is_transaction_form = issubclass(form.__class__, TransactionForm)
+    if is_transaction_form:
+        form.asset = asset
```

Comment:
Why do we need the asset to live in the form? It doesn't seem like we'd want to show this as a form element to the user, its not something they can change

Reply:
`form.asset` is used in the validation function for the _amount_ field, which is called from `form.is_valid()`. We don't control the parameters for the validation function so we have to give the form access to the non-field object somehow.

`asset` is not shown in the form, only attributes of type `Field` are displayed.